solidity/dashboard/src/pages/TokensPageContainer.jsx

```diff
@@ -479,10 +472,16 @@ const TokensPageContainerWithRedux = connect(
   mapDispatchToProps
 )(TokensPageContainer)
 
-const TokensPageContainerWithContext = () => (
+const TokensPageContainerWithContext = React.memo((props) => (
   <TokensPageContextProvider>
-    <TokensPageContainerWithRedux />
+    <TokensPageContainerWithRedux {...props} />
   </TokensPageContextProvider>
-)
+))
+
+TokensPageContainerWithContext.route = {
+  title: "Tokens",
+  path: "/tokens",
+  pages: [TokenOverviewPage, TokensPage, TokenGrantsPage],
+}
```

Comment:
How does this approach feel? Is it better/worse/neutral compared to what we had before?

Reply:
I like that abstraction and that we have a special type of component- `Page` component, that requires extra attributes and I think perfectly fits to our design.

Just wonder how could be problematic if we have a "special" type of page that will require something that we didn't predict in this solution. I mean I would like to avoid a situation where we need to pass a ton of props to the PageWrapper or other Routing components only to render this "special" page.